src/mame/drivers/a2600.cpp

```diff
@@ -42,6 +114,75 @@ void a2600_base_state::a2600_mem(address_map &map) // 6507 has 13-bit address sp
 }
 
 
+void a2600_pop_state::memory_map(address_map &map) // 6507 has 13-bit address space, 0x0000 - 0x1fff
+{
+	map(0x0000, 0x007f).mirror(0x0f00).rw(m_tia, FUNC(tia_video_device::read), FUNC(tia_video_device::write));
+	map(0x0080, 0x00ff).mirror(0x0d00).ram().share("riot_ram");
+#if USE_NEW_RIOT
+	map(0x0280, 0x029f).mirror(0x0d00).m("riot", FUNC(mos6532_t::io_map));
+#else
+	map(0x0280, 0x029f).mirror(0x0d00).rw("riot", FUNC(riot6532_device::read), FUNC(riot6532_device::write));
+#endif
+	map(0x0800, 0x0800).rw(FUNC(a2600_pop_state::rom_switch_r), FUNC(a2600_pop_state::rom_switch_w));
+	map(0x1000, 0x1fff).rw(m_bankdev, FUNC(address_map_bank_device::read8), FUNC(address_map_bank_device::write8));
+}
+
+
+void a2600_pop_state::banked_map(address_map &map)
+{
+	map(0x00000, 0x02ffff).rom().region("maincpu", 0);
+}
+
+
+//  read returns number of empty game rom slots
+uint8_t a2600_pop_state::rom_switch_r(offs_t offset)
+{
+	return 5;	// Max 47 games, 5 empty slots => 42 games
+}
+
+
+//	Rom switch
+void a2600_pop_state::rom_switch_w(offs_t offset, uint8_t data)
+{
+	m_bankdev->set_bank(data & 0x7f);
+	if (data & 0x80) {
+		// (re)start reset timer?
+		// Unknown what happens when multiple dipswitches are set
+		uint8_t a8 = m_a8->read();
+		attotime reset_time = attotime::never;
+		if (a8 & 0x08) {
+			// infinite
+			// enable reset by pressing game selection button when on infinite timer
+			m_game_select_button_timer->adjust(attotime::from_hz(60), 0, attotime::from_hz(60));
+		} else if (a8 & 0x04) {
+			// 30 seconds
+			reset_time = attotime::from_seconds(30);
+		} else if (a8 & 0x02) {
+			// 1 minute
+			reset_time = attotime::from_seconds(60);
+		} else if (a8 & 0x01) {
+			// 2 minutes
+			reset_time = attotime::from_seconds(120);
+		}
```

Comment:
The rest of this file is Allman style.

Reply:
Code style now matches the rest of the file